main.c

```diff
@@ -157,6 +157,42 @@ static void read_config(const char *fname)
                 fatal_error("%s: syntax error on line %i\n", fname, lineNum);
             }
         }
+        else if (strcmp(tokens[0], "thumb_label") == 0)
+        {
+            int addr;
+
+            if (sscanf(tokens[1], "%i", &addr) == 1)
+            {
+                if (strlen(tokens[2]) != 0)
+                    name = dup_string(tokens[2]);
+                int label = disasm_add_label(addr, LABEL_THUMB_CODE, name);
+                if(!gLabels)
+                	fatal_error("%s: failed to add label on line %i, gLabels is NULL\n", fname, lineNum); // shouldn't happen
```

Comment:
I don't see why you need this. `disasm_add_label` will call `fatal_error` directly when the allocation fails.